lib/util/itinerary.js

```diff
@@ -47,3 +48,29 @@ export function itineraryCanBeMonitored (itinerary) {
 export function getMinutesUntilItineraryStart (itinerary) {
   return moment(itinerary.startTime).diff(moment(), 'minutes')
 }
+
+/**
+ * Gets the first transit leg of the given itinerary, or null if none found.
+ */
+function getFirstTransitLeg (itinerary) {
+  return itinerary?.legs?.find(leg => leg.transitLeg)
+}
+
+/**
+ * Returns the set of monitored days that will be initially shown to the user
+ * for the given transit itinerary (itinerary with transit leg).
+ * @param itinerary The itinerary from which the default monitored days are extracted.
+ * @returns ['monday' thru 'friday'] if itinerary happens on a weekday(*),
+ *          ['saturday', 'sunday'] if itinerary happens on a saturday/sunday(*).
+ * (*) The first transit leg will be used to make the determination.
+ */
+export function getTransitItineraryDefaultMonitoredDays (itinerary) {
+  const firstTransitLeg = getFirstTransitLeg(itinerary)
+  // Assume a transit leg exists.
+  if (!firstTransitLeg) return null
```

Comment:
Is returning null helpful here? I think a better alternative would be to just check the start time of the trip and not worry about the service day if this is not a transit itinerary.

Reply:
Sure, updated in https://github.com/opentripplanner/otp-react-redux/pull/385/commits/4eaa00046b5dbcda886968309c4452f49cda1237.